documentation/cubic-pos-article.md

```diff
@@ -0,0 +1,155 @@
+
+**Important parameters in the Namada PoS model**
+- Duration of a block: ~ 5 - 10 seconds
+- Duration of an epoch: ~ 1 day
+- Pipeline offset length: 2 epochs
+- Unbonding offset length: 21 epochs
+
+Below, we will review the rules around important PoS actions
+- Bonding tx submitted at epoch `n` initiates changes at epoch `n + pipeline_offset (n+2)`
+	- Bond becomes active, validator bonded stake and validator set is updated
+- Unbonding tx submitted at epoch `n`
+	- Updates validator set with decreased validator voting power at epoch `n + pipeline_offset (n+2)`
+	- Amount of the unbond is withdrawable at epoch `n + unbonding_offset + pipeline_offset (n+23)`
+		- TODO: check over this w Tomas
+- Redelegation tx submitted at epoch `n`
+	- Only support 100% token amount redelegated
+	- TODO: more details needed here perhaps?
+
+**Validator Sets**
+
+Namada validators are classified into one of three sets:
+- **Consensus:** validators that are active in the consensus voting mechanism, chosen to be the `max_validators` largest validators weighted by their voting power. They are the only validators who can earn block rewards from the PoS inflation.
+- **Below capacity:** validators whose bonded stake is less than the *consensus* validators but above some small threshold `min_validator_stake`.
+- **Below threshold:** all other validators with stake below `min_validator_stake`.
+
+The *consensus* and *below capacity* validator sets are explicitly held in storage, ordered by their bonded stake, and are updated at every epoch boundary. However, no ordered set of *below threshold* validators is kept in storage. This validator set construction and the `min_validator_stake` threshold exist primarily to prevent the possibility of significant slowing of PoS system updates due to spam validator account creation. Thus, unbounded iteration when updating and validating PoS system state changes is avoided. Conversely, the *below capacity* validator set must be kept in storage, ordered by bonded stake, in order to properly process validator set changes in which validators may drop out or enter the *consensus* validator set.
+
+**Important validator set configuration values, mutable via governance:**
+- `max_validators` = 100
+- `min_validator_stake` = 1 NAM
+
+# Cubic Slashing
+
+As in many other PoS blockchains, slashing is applied as a way to punish misbehaving validators. If a validator is detected to have misbehaved then portions of the validator's bonded stake will be slashed or seized. This propagates down to the stake of delegators associated to the misbehaving validator as well. Validator infractions that are subject to slashing include:
+
+- proposing an invalid block
+- signing an invalid block
+- proposing two different blocks at the same height
+- double-signing a block
+
+Ultimately, slashed tokens are sent to a slash pool fund, which...
```

Comment:
the fund could be spent by governance to refund in the case of accidental misbehaviour